conan/external_tools/vswhere_test.py

```diff
@@ -0,0 +1,169 @@
+        install_path = tools.vs_installation_path("15", preference=["BuildTools", "Community", "Professional", "Enterprise"])
+        self.assertNotIn("Community", install_path)
+        self.assertNotIn("Professional", install_path)
+        self.assertNotIn("Enterprise", install_path)
+        self.assertIn("BuildTools", install_path)
+
+        install_path = tools.vs_installation_path("15", preference=["Professional", "Enterprise", "Community"])
+        self.assertNotIn("BuildTools", install_path)
+        self.assertNotIn("Professional", install_path)
+        self.assertNotIn("Enterprise", install_path)
+        self.assertIn("Community", install_path)
+
+        # Preference order by env var
+        with(tools.environment_append({"CONAN_VS_INSTALLATION_PREFERENCE":"BuildTools, Community,Professional, Enterprise"})):
+            install_path = tools.vs_installation_path("15")
+            self.assertNotIn("Community", install_path)
+            self.assertNotIn("Professional", install_path)
+            self.assertNotIn("Enterprise", install_path)
+            self.assertIn("BuildTools", install_path)
+        
+        with(tools.environment_append({"CONAN_VS_INSTALLATION_PREFERENCE":"Professional, Enterprise,Community"})):
+            install_path = tools.vs_installation_path("15")
+            self.assertNotIn("BuildTools", install_path)
+            self.assertNotIn("Professional", install_path)
+            self.assertNotIn("Enterprise", install_path)
+            self.assertIn("Community", install_path)
+
+    def vvcars_command_test(self):
+        fake_settings = settings.Settings({"os":"Windows", "arch": "x86_64"})
+
+        # preference order with VS 15
+        with(tools.environment_append({"CONAN_VS_INSTALLATION_PREFERENCE":"BuildTools, Community,Professional, Enterprise"})):
+            command = tools.vcvars_command(settings=fake_settings, compiler_version="15")
+            self.assertNotIn("Community", command)
+            self.assertIn("VC/Auxiliary/Build/vcvarsall.bat", command)
+            self.assertIn("Microsoft Visual Studio\\2017\\BuildTools", command)
+            self.assertIn("VSCMD_START_DIR", command)
+
+        with(tools.environment_append({"CONAN_VS_INSTALLATION_PREFERENCE":"Professional, Enterprise,Community"})):
+            command = tools.vcvars_command(settings=fake_settings, compiler_version="15")
+            self.assertNotIn("BuildTools", command)
+            self.assertIn("VC/Auxiliary/Build/vcvarsall.bat", command)
+            self.assertIn("Microsoft Visual Studio\\2017\\Community", command)
+            self.assertIn("VSCMD_START_DIR", command)
+
+        # With VS 14 order of preference does not apply
+        command = tools.vcvars_command(settings=fake_settings, compiler_version="14")
+        self.assertNotIn("VSCMD_START_DIR", command)
+        self.assertIn("VC/vcvarsall.bat", command)
+        self.assertIn("Microsoft Visual Studio 14.0\\", command)
+
+    def build_test(self):
```

Comment:
probably this is already very very tested in the conan regular suite, right?

Reply:
Would we want maybe to force building with the build tools instead? (env var) and test that it is really happening and it is working also?

Reply:
Yes, I think this could be better! Maybe just use test client to build Hello and check cmake output to assert it is using BuildTools for building